src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/MetaDataQueriesIT.java

```diff
@@ -290,6 +290,34 @@ public void describeSingleIndex() throws IOException {
         assertThat(row.get(5), not(equalTo(JSONObject.NULL)));
     }
 
+    @Test
+    public void describeSingleIndexWithObjectFieldShouldPass() throws IOException {
+        JSONObject response =
+                executeQuery(String.format("DESCRIBE TABLES LIKE %s", TestsConstants.TEST_INDEX_GAME_OF_THRONES));
+
+        // Schema for DESCRIBE is filled with a lot of fields that aren't used so only the important
+        // ones are checked for here
+        String[] fields = {"TABLE_NAME", "COLUMN_NAME", "TYPE_NAME"};
+        checkContainsColumns(getSchema(response), fields);
+
+        JSONArray dataRows = getDataRows(response);
+        assertThat(dataRows.length(), greaterThan(0));
+        assertThat(dataRows.getJSONArray(0).length(), equalTo(DESCRIBE_FIELD_LENGTH));
+
+        /*
+         * Assumed indices of fields in dataRows based on "schema" output for DESCRIBE given above:
+         * "TABLE_NAME"  : 2
+         * "COLUMN_NAME" : 3
+         * "TYPE_NAME"   : 5
+         */
+        for (int i = 0; i < dataRows.length(); i++) {
+            JSONArray row = dataRows.getJSONArray(i);
+            assertThat(row.get(2), equalTo(TestsConstants.TEST_INDEX_GAME_OF_THRONES));
+            assertThat(row.get(3), not(equalTo(JSONObject.NULL)));
+            assertThat(row.get(5), not(equalTo(JSONObject.NULL)));
```

Comment:
We should be checking `TYPE_NAME` to be `object`, instead of checking its `not null`.

Also can you test how this would impact JDBC driver.

Reply:
I'm thinking of adding IT with JDBC driver involved later because our JDBC driver is already added as test dependency in Gradle. However, the problem is we don't have nightly build so it's difficult to test with latest changes in JDBC driver. The comparison test has same issue: https://github.com/opendistro-for-elasticsearch/sql/issues/339#issue-547727767

Reply:
Done. Verify the each row.
For testing with JDBC driver, I agree with chen's idea to do integration test based on nightly build pipeline.